### PR TITLE
get user's completed challenges route

### PR DESCRIPTION
**Pull request by AadharSharma7**
What were your changes

Added stuff to users.js and completedRecipesModel.js.

What specifically should be checked

Check if the query looks fine in completedRecipesModel, and if the route looks fine in users.js.

### CODE REVIEW

File: source/backend/database/models/completedRecipesModel.js [hunk 1]
```diff
@@ -13,4 +13,18 @@ async function getCompletedRecipes(userId){
...
+async function getCompletedChallenges(userId, challenge) {
+  const completedChallenges = await db('completedRecipes')
+    .join('users', 'completedRecipes.userId', 'users.userId')
+    .join('recipes', 'completedRecipes.recipeId', 'recipes.recipeId')
+    .select('recipes.*')
+    .where({userId, challenge});
+  return completedChallenges; 
+}
+
+module.exports = { getCompletedRecipes, getCompletedChallenges};
```
Comment (demyinn00) [inline source/backend/database/models/completedRecipesModel.js:30]: We talked about this earlier, but i think this makes sense. We'll have Thomas take a look at this to make sure, but I think it's good.

File: source/backend/routes/users.js [hunk 2]
```diff
@@ -35,6 +35,25 @@ router.get('/savedRecipes', async (req, res) => {
...
+  } catch (err) {
+    console.error(err);
+    return res.status(503).json({
+      message: 'Failed to get user information due to'
+        + 'internal server error',
+      err,
+    });
+  }
+});
+
```
Comment (demyinn00) [inline source/backend/routes/users.js:56]: The request endpoint is up to us. I believe when we do something like `http://localhost:3000/users/challenge/Spicy+Sips` we need to consider how it will be formatted. I guess this is something we need a bit more clarification on, but for now I think we can assume that every space will be separated by a `+` since that's generally the convention. 

TL;DR
req.params -> Spicy+Sips
it's also possible we get an invalid challenge, so add error handling to verify the challenge exists. 
Psuedocode: 
challenge = req.challenge
parse challenge -> 'Spicy+Sips' -> 'Spicy Sips'
if that's not a valid challenge: return status 400 (bad request)
otherwise: return the recipes from that challenge

File: node_modules/.package-lock.json [hunk 3]
```diff
@@ -4156,6 +4156,20 @@
...
       "dev": true
     },
+    "node_modules/fsevents": {
+      "version": "2.3.2",
+      "resolved": "https://registry.npmjs.org/fsevents/-/fsevents-2.3.2.tgz",
+      "integrity": "sha512-xiqMQR4xAeHTuB9uWm+fFRcIOgKBMiOBP+eXiyT7jsgVCq1bkVygt00oASowB7EdtpOHaaPgKt812P9ab+DDKA==",
+      "dev": true,
+      "hasInstallScript": true,
+      "optional": true,
+      "os": [
```
Comment (joecai6) [inline node_modules/.package-lock.json:4166]: There are some changes in the config files from dexie that are part of the PR. Are these changes needed? If not, you can remove these change from the PR.

Comment (AadharSharma7) [reply node_modules/.package-lock.json:4166]: I don't think these changes are needed, my bad.

File: source/backend/routes/users.js (hunk 2 above)
Comment (AadharSharma7) [reply source/backend/routes/users.js]: Done. Let me know if there are any issues.
